transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -85,11 +88,22 @@ public EmbeddedChannel(ChannelId channelId) {
      *
      * @param handlers the {@link ChannelHandler}s which will be add in the {@link ChannelPipeline}
      */
-    public EmbeddedChannel(final ChannelHandler... handlers) {
+    public EmbeddedChannel(ChannelHandler... handlers) {
         this(EmbeddedChannelId.INSTANCE, handlers);
     }
 
     /**
+     * Create a new instance with the pipeline initialized with the specified handlers.
+     *
+     * @param hasDisconnect {@code true} if this {@link Channel} will handle {@link #disconnect()} and {@link #close()}
+     *                      differently, {@code false} otherwise.
```

Comment:
because we did non "different" things before and I can not break this. Also this is what is done with `SocketChannel` impls. So most of the times people may expect it